src/button/button-group.scss

```diff
@@ -9,25 +9,21 @@
 
   $childSelector: 'input, button'; // target nested inputs and buttons
 
-  > * {
+  .iui-button-group-item {
     display: flex;
+    position: relative;
 
-    #{$childSelector} {
-      position: relative;
+    &:focus-within {
+      z-index: 2;
+    }
 
+    > * {
       &:hover {
         z-index: 1;
       }
 
-      &:focus {
-        z-index: 2;
-
-        &:where(:not(:focus-visible):not(:hover)) {
-          z-index: revert;
-        }
-      }
-
-      &:disabled {
+      &:disabled,
+      &[aria-disabled='true'] {
```

Comment:
also thought, do we everywhere check `aria-disabled` where we check for disabled status? Personally, do not remember this in other places. If not, should we have this recorded to do one day?

Reply:
I added this to be future-proof. I think I should also add `iui-disabled` because we use that in labeled-input

Reply:
`aria-disabled` is only present in `button-group.scss` as of this PR.  I think it makes sense to include this in other places.  Here is a list of where we are referencing `:disabled` and could probably include it:
- /button/borderless.scss
- /button/button.scss
- /button/cta.scss
- /button/default.scss
- /button/high-visibility.scss
- /button/split-menu.scss
- /header/header.scss
- /inputs/checkbox-radio.scss
- /inputs/checkbox.scss
- /inputs/radio-tile.scss
- /side-navigation/side-navigation.scss
- /toggle-switch/toggle-switch.scss

I also agree with including `.iui-disabled`, that as well as the aria method may help us achieve `:focus` on a disabled button which has been requested in the past.

Reply:
Now seems we ended up in inconsistencies.. one place we target disabled and aria-disabled, but on line 50 it is only disabled state...
That part also has too deep nesting for our future refactors..

Reply:
Why `aria-disabled` is even needed here?

Reply:
> Now seems we ended up in inconsistencies

made the disabled selector consistent by adding `iui-disabled` and storing it in a variable

> Why aria-disabled is even needed here?

See discussion above. It will help with #391

Reply:
Didnt want to check anything about the nesting? Anyway you are changing a bunch of stuff here, wouldnt need to come back